doordeck-sdk-swift/InitializeDoordeck.swift

```diff
@@ -237,118 +218,52 @@ public class Doordeck {
     /// - Parameter token: new AuthTokenClass from the Host application
     func updateAuthToken (_ token: AuthTokenClass?) {
         guard let tokenTemp: AuthTokenClass = token else { return }
-        self.token = tokenTemp
-        let header = Header().createSDKAuthHeader(.v1, token: self.token)
-        self.apiClient = APIClient(header, token: self.token)
-        self.sodium = SodiumHelper(self.token)
+        doordeckSDK.contextManager().setCloudAuthToken(token: tokenTemp.getToken())
     }
     
-    /// Return Certificate chain when SDK is being used in different format.
-    ///
-    /// - Parameter Certificate chain
-    func getCertificateChain() -> CertificateChainClass? {
-        guard let chainTemp = self.chain else {
-            return nil
-        }
-        if sdk == false {
-            return chainTemp
+    var currentState: State {
+        doordeckSDK.contextManager().getCertificateChain()
```

Comment:
What does the certificate chain retrieval do here?

Reply:
I'm glad I asked you for review :D
fixing it now